CMakeLists.txt

```diff
@@ -16,24 +16,76 @@ project(
     LANGUAGES CXX
 )
 
-#---- C++ standard that we need (at least).
-set(CMAKE_CXX_STANDARD          20   CACHE BOOL "" FORCE)
-set(CMAKE_CXX_STANDARD_REQUIRED True CACHE BOOL "" FORCE)
+#---- Options.
+option(KOKKOSUTILS_ENABLE_TESTS "Enable testing" ON)
+option(KOKKOSUTILS_ENABLE_DOC   "Enable documentation" ON)
 
 #---- Find Kokkos.
-find_package(
-    Kokkos
-    ${KOKKOS_VERSION}
-    REQUIRED
-)
+if(NOT TARGET Kokkos::kokkoscore)
+    find_package(
+        Kokkos
+        ${KOKKOS_VERSION}
+        REQUIRED
+    )
+endif()
 
 #---- Find Google Test.
-find_package(
-    GTest
-    ${GOOGLETEST_VERSION}
-    CONFIG
-    REQUIRED
-)
+if(NOT TARGET GTest::gtest_main)
+    find_package(
+        GTest
+        ${GOOGLETEST_VERSION}
+        CONFIG
+        REQUIRED
+    )
+    endif()
 
 #---- Documentation.
-add_subdirectory(docs)
+if(KOKKOSUTILS_ENABLE_DOC)
+    #--- Find Doxygen.
+    if(NOT TARGET Doxygen::doxygen)
+        find_package(
+            Doxygen
+            ${DOXYGEN_VERSION}
+            REQUIRED
+        )
+    endif()
```

Comment:
I would still look for `Doxygen` in the subdirectory, not in the main `CMakeLists.txt` :wink: